worker-db-ddl.sql

```diff
@@ -38,26 +40,85 @@ CREATE TYPE cqc."WorkerDisability" AS ENUM (
+	"MainJobStartDateSavedBy" VARCHAR(120) NULL,
+	"MainJobStartDateChangedBy" VARCHAR(120) NULL,
+	"NationalInsuranceNumberValue" VARCHAR(13) NULL,
+	"NationalInsuranceNumberSavedAt" TIMESTAMP NULL,
+	"NationalInsuranceNumberChangedAt" TIMESTAMP NULL,
+	"NationalInsuranceNumberSavedBy" VARCHAR(120) NULL,
+	"NationalInsuranceNumberChangedBy" VARCHAR(120) NULL,
+	"DateOfBirthValue" DATE NULL,
+	"DateOfBirthSavedAt" TIMESTAMP NULL,
+	"DateOfBirthChangedAt" TIMESTAMP NULL,
+	"DateOfBirthSavedBy" VARCHAR(120) NULL,
+	"DateOfBirthChangedBy" VARCHAR(120) NULL,
+	"PostcodeValue" VARCHAR(8) NULL,
+	"PostcodeSavedAt" TIMESTAMP NULL,
+	"PostcodeChangedAt" TIMESTAMP NULL,
+	"PostcodeSavedBy" VARCHAR(120) NULL,
+	"PostcodeChangedBy" VARCHAR(120) NULL,
+	"DisabilityValue" cqc."WorkerDisability" NULL,
+	"DisabilitySavedAt" TIMESTAMP NULL,
+	"DisabilityChangedAt" TIMESTAMP NULL,
+	"DisabilitySavedBy" VARCHAR(120) NULL,
+	"DisabilityChangedBy" VARCHAR(120) NULL,
+	"GenderValue" cqc."WorkerGender" NULL,
+	"GenderSavedAt" TIMESTAMP NULL,
+	"GenderChangedAt" TIMESTAMP NULL,
+	"GenderSavedBy" VARCHAR(120) NULL,
+	"GenderChangedBy" VARCHAR(120) NULL,
 	created TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW(),
 	updated TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW(),	-- note, on creation of record, updated and created are equal
+	updatedby VARCHAR(120) NOT NULL,
     CONSTRAINT "Worker_Establishment_fk" FOREIGN KEY ("EstablishmentFK") REFERENCES cqc."Establishment" ("EstablishmentID") MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
-	CONSTRAINT "Worker_Job_mainjob_fk" FOREIGN KEY ("MainJobFK") REFERENCES cqc."Job" ("JobID") MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
-	CONSTRAINT "Worker_WorkerUID_unq" UNIQUE ("WorkerUID", "EstablishmentFK")
+	CONSTRAINT "Worker_Job_mainjob_fk" FOREIGN KEY ("MainJobFKValue") REFERENCES cqc."Job" ("JobID") MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+	CONSTRAINT "Worker_WorkerUID_unq" UNIQUE ("WorkerUID")
 );
 
 CREATE UNIQUE INDEX "Worker_WorkerUID" on cqc."Worker" ("WorkerUID");
 CREATE INDEX "Worker_EstablishmentFK" on cqc."Worker" ("EstablishmentFK");
 
-ALTER TABLE cqc."Worker" ADD COLUMN "Gender" cqc."WorkerGender" NULL;
-ALTER TABLE cqc."Worker" ADD COLUMN "Disability" cqc."WorkerDisability" NULL;
\ No newline at end of file
+-- change auditting
+CREATE TYPE cqc."AuditChangeType" AS ENUM (
+	'created',
+	'updated',
+	'saved',
+	'changed'
+);
+CREATE TABLE IF NOT EXISTS cqc."WorkerAudit" (
+	"ID" SERIAL NOT NULL PRIMARY KEY,
+	"WorkerFK" INTEGER NOT NULL,
+	"Username" VARCHAR(120) NOT NULL,
+	"When" TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW(),
+	"EventType" cqc."AuditChangeType" NOT NULL,
+	"PropertyName" VARCHAR(100) NULL,
+	"ChangeEvents" JSONB NULL,
+	CONSTRAINT "WorkerAudit_Worker_fk" FOREIGN KEY ("WorkerFK") REFERENCES cqc."Worker" ("ID") MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION
+);
+CREATE INDEX "WorkerAudit_WorkerFK" on cqc."WorkerAudit" ("WorkerFK");
```

Comment:
Hi Warren

I would recommend the object naming convention should be discussed and defined to a unified agreed format.
For example for index object we can follow something like TableName_ColumnOfIndex_IDX. 

Though this doesn't affect any functionality however its easy to recognize by name itself what object is about looking at its name.

Rest All looks good. I am approving and merging this however we can define all the standards in forthcoming code reviews, if possible.


Thanks

Shakir